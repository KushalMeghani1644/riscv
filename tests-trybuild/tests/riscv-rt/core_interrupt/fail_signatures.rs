#[riscv_rt::core_interrupt(riscv::interrupt::Interrupt::SupervisorSoft)]
fn my_interrupt(code: usize) {}

#[riscv_rt::core_interrupt(riscv::interrupt::Interrupt::SupervisorTimer)]
fn my_other_interrupt() -> usize {}

#[riscv_rt::core_interrupt(riscv::interrupt::Interrupt::SupervisorExternal)]
async fn my_async_interrupt() {}

fn main() {}
