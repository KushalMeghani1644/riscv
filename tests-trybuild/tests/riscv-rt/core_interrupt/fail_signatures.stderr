error: `#[core_interrupt]` function must have signature `[unsafe] fn() [-> !]`
 --> tests/riscv-rt/core_interrupt/fail_signatures.rs:2:1
  |
2 | fn my_interrupt(code: usize) {}
  | ^^

error: `#[core_interrupt]` function must have signature `[unsafe] fn() [-> !]`
 --> tests/riscv-rt/core_interrupt/fail_signatures.rs:5:1
  |
5 | fn my_other_interrupt() -> usize {}
  | ^^

error: `#[core_interrupt]` function must have signature `[unsafe] fn() [-> !]`
 --> tests/riscv-rt/core_interrupt/fail_signatures.rs:8:1
  |
8 | async fn my_async_interrupt() {}
  | ^^^^^
